From 361ec5d439b290e659cec63071f499fc0104401c Mon Sep 17 00:00:00 2001
From: Tomas Chvatal <tomas.chvatal@gmail.com>
Date: Wed, 24 Jul 2013 22:38:55 +0200
Subject: [PATCH] Fix mixing C and C++ in templates

This kills the favorite error for some gcc versions about C in the
templates.

*snip*
/usr/lib/gcc/x86_64-pc-linux-gnu/4.6.3/include/g++-v4/bits/basic_string.tcc:1068:3:
error: template with C linkage
make[2]: *** [zypp/CMakeFiles/zypp.dir/parser/xml/Node.cc.o] Error 1
*snip*

Basically we don't need to put this stuff here as it is done in the
headers themselves, so just prune the problematic code.
---
 zypp/parser/xml/Node.cc   | 3 ---
 zypp/parser/xml/Reader.cc | 3 ---
 2 files changed, 6 deletions(-)

diff --git a/zypp/parser/xml/Node.cc b/zypp/parser/xml/Node.cc
index e3ac4cd..c90b8e1 100644
--- a/zypp/parser/xml/Node.cc
+++ b/zypp/parser/xml/Node.cc
@@ -9,11 +9,8 @@
 /** \file zypp/parser/xml/Reader.cc
  *
 */
-extern "C"
-{
 #include <libxml/xmlreader.h>
 #include <libxml/xmlerror.h>
-}
 
 #include <iostream>
 
diff --git a/zypp/parser/xml/Reader.cc b/zypp/parser/xml/Reader.cc
index 4e514f4..d06be45 100644
--- a/zypp/parser/xml/Reader.cc
+++ b/zypp/parser/xml/Reader.cc
@@ -9,11 +9,8 @@
 /** \file zypp/parser/xml/Reader.cc
  *
 */
-extern "C"
-{
 #include <libxml/xmlreader.h>
 #include <libxml/xmlerror.h>
-}
 
 #include <iostream>
 
-- 
1.8.1.5

